lab5/app/src/main/java/com/app/lab5/HttpURLConnectionActivity.java

```diff
@@ -0,0 +1,243 @@
+        {
+            @Override
+            public void run() {
+                // Maintain http url connection.
+                HttpURLConnection httpConn = null;
+
+                // Read text input stream.
+                InputStreamReader isReader = null;
+
+                // Read text into buffer.
+                BufferedReader bufReader = null;
+
+                // Save server response text.
+                StringBuffer readTextBuf = new StringBuffer();
+
+                try {
+                    // Create a URL object use page url.
+                    URL url = new URL(reqUrl);
+
+                    // Open http connection to web server.
+                    httpConn = (HttpURLConnection)url.openConnection();
+
+                    // Set http request method to get.
+                    httpConn.setRequestMethod(REQUEST_METHOD_GET);
+
+                    // Set connection timeout and read timeout value.
+                    httpConn.setConnectTimeout(10000);
+                    httpConn.setReadTimeout(10000);
+
+                    // Get input stream from web url connection.
+                    InputStream inputStream = httpConn.getInputStream();
+
+                    // Create input stream reader based on url connection input stream.
+                    isReader = new InputStreamReader(inputStream);
+
+                    // Create buffered reader.
+                    bufReader = new BufferedReader(isReader);
+
+                    // Read line of text from server response.
+                    String line = bufReader.readLine();
+
+                    // Loop while return line is not null.
+                    while(line != null)
+                    {
+                        // Append the text to string buffer.
+                        readTextBuf.append(line);
+
+                        // Continue to read text line.
+                        line = bufReader.readLine();
+                    }
+
+                    // Send message to main thread to update response text in TextView after read all.
+                    Message message = new Message();
+
+                    // Set message type.
+                    message.what = REQUEST_CODE_SHOW_RESPONSE_TEXT;
+
+                    // Create a bundle object.
+                    Bundle bundle = new Bundle();
+                    // Put response text in the bundle with the special key.
+                    bundle.putString(KEY_RESPONSE_TEXT, readTextBuf.toString());
+                    // Set bundle data in message.
+                    message.setData(bundle);
+                    // Send message to main thread Handler to process.
+                    uiUpdater.sendMessage(message);
+                }catch(MalformedURLException ex)
+                {
+                    Log.e(TAG_HTTP_URL_CONNECTION, ex.getMessage(), ex);
+                }catch(IOException ex)
+                {
+                    Log.e(TAG_HTTP_URL_CONNECTION, ex.getMessage(), ex);
```

Comment:
To avoid duplication of code you can just have a single catch statement with the exceptions separated by |